cli/src/cmd/forge/script/sequence.rs

```diff
@@ -0,0 +1,379 @@
+        sig: &str,
+        target: &ArtifactId,
+        config: &Config,
+        chain_id: u64,
+    ) -> eyre::Result<Self> {
+        let path = ScriptSequence::get_path(&config.broadcast, sig, target, chain_id)?;
+
+        Ok(ScriptSequence {
+            transactions,
+            receipts: vec![],
+            pending: vec![],
+            path,
+            timestamp: SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .expect("Wrong system time.")
+                .as_secs(),
+            libraries: vec![],
+        })
+    }
+
+    pub fn load(
+        config: &Config,
+        sig: &str,
+        target: &ArtifactId,
+        chain_id: u64,
+    ) -> eyre::Result<Self> {
+        let file = std::fs::read_to_string(ScriptSequence::get_path(
+            &config.broadcast,
+            sig,
+            target,
+            chain_id,
+        )?)?;
+        serde_json::from_str(&file).map_err(|e| e.into())
+    }
+
+    pub fn save(&mut self) -> eyre::Result<()> {
+        if !self.transactions.is_empty() {
+            self.timestamp = SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs();
+
+            let path = self.path.to_str().wrap_err("Invalid path.")?;
+
+            //../run-latest.json
+            serde_json::to_writer(BufWriter::new(std::fs::File::create(path)?), &self)?;
+            //../run-[timestamp].json
+            serde_json::to_writer(
+                BufWriter::new(std::fs::File::create(
+                    path.replace("latest.json", &format!("{}.json", self.timestamp)),
+                )?),
+                &self,
+            )?;
+
+            println!("\nTransactions saved to: {path}\n");
+        }
+
+        Ok(())
+    }
+
+    pub fn add_receipt(&mut self, receipt: TransactionReceipt) {
+        self.receipts.push(receipt);
+    }
+
+    pub fn sort_receipts(&mut self) {
+        self.receipts.sort_by(|a, b| {
+            let ablock = a.block_number.unwrap();
+            let bblock = b.block_number.unwrap();
+            if ablock == bblock {
+                a.transaction_index.cmp(&b.transaction_index)
+            } else {
+                ablock.cmp(&bblock)
+            }
+        });
+    }
+
+    pub fn add_pending(&mut self, index: usize, tx_hash: TxHash) {
+        if !self.pending.contains(&tx_hash) {
+            self.transactions[index].hash = Some(tx_hash);
+            self.pending.push(tx_hash);
+        }
+    }
+
+    pub fn remove_pending(&mut self, tx_hash: TxHash) {
+        self.pending.retain(|element| element != &tx_hash);
+    }
+
+    pub fn add_libraries(&mut self, libraries: Libraries) {
+        self.libraries = libraries
+            .libs
+            .iter()
+            .flat_map(|(file, libs)| {
+                libs.iter().map(|(name, address)| {
+                    format!("{}:{}:{}", file.to_string_lossy(), name, address)
+                })
+            })
+            .collect();
```

Comment:
nice

Reply:
Looks good - but let's also pull this in a helper function on `Libraries`